gzip.go

```diff
@@ -129,11 +129,16 @@ func (w *GzipResponseWriter) startGzip() (int, error) {
 
 	// Flush the buffer into the gzip reponse.
 	n, err := w.gw.Write(w.buf)
-	// Empty the buffer.
-	w.buf = nil
 
-	// Return the numbers of bytes writen and the error if any.
-	return n, err
+	// This should never happen (per io.Writer docs), but if the write didn't
```

Comment:
Well, [the interface](https://golang.org/pkg/io/#Writer) says

> Write must return a non-nil error if it returns n < len(p).

but that doesn't mean that writers will actually do it. So rather than pass along the violation, I replaced it with an error, like `io.Copy` does. One could reasonably argue that we should panic instead, since this state indicates that something is seriously fucked, but that seemed a little excessive.